staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchNameProvider.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2015-2018 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging.dao.filesystem;
+
+import com.github.cafdataprocessing.services.staging.BatchId;
+import com.github.cafdataprocessing.services.staging.utils.ServiceIdentifier;
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+
+public final class BatchNameProvider
+{
+    private static final String SUBBATCH_FILE_SUFFIX = "-json.batch";
+
+    public static String getBatchDirectoryName(final BatchId batchId)
+    {
+        return getCurrentTimeAsString()
+            .concat("-").concat("" + Thread.currentThread().getId())
+            .concat("-").concat(ServiceIdentifier.getServiceId())
+            .concat("-").concat(batchId.getValue());
+    }
+
+    public static String getSubBatchName()
+    {
+        return getCurrentTimeAsString().concat(SUBBATCH_FILE_SUFFIX);
+    }
+
+    private static String getCurrentTimeAsString()
+    {
+        return cleanseTime(Instant.now().toString());
+    }
+
+    public static long getFileCreationTime(final String fileName)
+    {
+        final String time = fileName.substring(0, 22);
+
+        try {
+            System.out.println("File creation time file name: " + time);
+            return Instant.parse(reverseCleanseTimeString(time)).toEpochMilli();
```

Comment:
What timezone will this use?

Reply:
UTC, that is why we have to do the reverse cleanse, to have the UTC time in the file name we need to remove the colons from the time. To parse it back we have to put them back in before parsing.